src/main/java/mate/academy/bookstore/repository/book/specification/PriceSpecificationProvider.java

```diff
@@ -1,23 +1,29 @@
 package mate.academy.bookstore.repository.book.specification;
 
+import java.util.Optional;
+import mate.academy.bookstore.dto.BookSearchParametersDto;
 import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.SpecificationProvider;
 import mate.academy.bookstore.repository.book.BookSearchParameter;
 import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Component;
 
 @Component
-public class PriceSpecificationProvider extends AbstractSpecificationProvider<Book> {
-    private static final int MIN_PRICE = 0;
-    private static final int MAX_PRICE = 1;
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
+    private static final int DEFAULT_MIN_PRICE = 0;
+    private static final int DEFAULT_MAX_PRICE = Integer.MAX_VALUE;
```

Comment:
A point for optimizing the queries. Would be nice to not send the default params when we did not receive them in DTO because they are redundant, like:
* if we do not receive `minPrice` & `maxPrice` - query should not have that price filtration at all
* if we receive only `minPrice` - query should be just `where price > minPrice`
* if we receive only `maxPrice` - query should be just `where price < minPrice`